### PR TITLE
feat: maintain cursor position despite formatting

### PR DESCRIPTION
**Pull request by telemachus**

### CODE REVIEW

**Comment by telemachus** [comment]
Per conversation in #39, I've added a configuration option. I also added the new option to the README and put a note about the option into the docs.

I'm happy to make adjustments if you want, and you should feel free (of course!) to adjust the code to suit your style or how you want the option to work. Thanks for considering this. Let me know if you'd like me to make any changes.

(Sorry for the double pull request. I had to make a quick edit because of a dumb mistake.)

File: README.md [hunk 1]
```diff
@@ -104,6 +104,8 @@ require('go').setup({
     -- formatter: goimports, gofmt, gofumpt
     formatter = 'goimports',
     -- test flags: -count=1 will disable cache
```
Comment (crispgm) [inline README.md]: Please put the new option above this line, because this comments the following options :)

**Comment by crispgm** [comment]
Thanks for the PR. There is a minor issue. Otherwise, it's perfect.

**Review by crispgm — CHANGES_REQUESTED** [review]

File: lua/go/config.lua [hunk 2]
```diff
@@ -12,6 +12,8 @@ M.options = {
     lint_prompt_style = 'qf',
     -- formatter: goimports, gofmt, gofumpt
     formatter = 'goimports',
+    -- maintain cursor position after formatting loaded buffer
+    maintain_cursor_pos = true,
```
Comment (crispgm) [inline lua/go/config.lua]: I think it better to change default to `false`, because it's new feature somehow breaking.

**Review by crispgm — CHANGES_REQUESTED** [review]

**Comment by telemachus** [comment]
Thanks for the review. I made the two changes you asked for.

**Comment by crispgm** [comment]
LGTM but you may have to resolve the conflicts.

**Comment by telemachus** [comment]
> LGTM but you may have to resolve the conflicts.

@crispgm I'm sorry, but I'm not sure how to do that.

This is the conflict:

```diff
<<<<<<< maintain-cursor-pos
        maintain_cursor_pos = false,
=======
        -- test flags: -count=1 will disable cache
>>>>>>> main
```

I don't see what I can do from my end to fix that,
especially since I have to force push to update this
pull request.

Any suggestions?

**Comment by telemachus** [comment]
> LGTM but you may have to resolve the conflicts.

Okay, I think I've done that. Let me know if you need
any other changes.

**Comment by crispgm** [comment]
> 

Looks fine. I will fix some minor style issue and then merge.

**Comment by crispgm** [comment]
Merged! Thanks very much :) @telemachus

**Pull request merged**